src/helix/endpoints/channels/get_ad_schedule.rs

```diff
@@ -0,0 +1,228 @@
+    /// The UTC timestamp when the broadcaster will gain an additional snooze, in RFC3339 format.
+    pub snooze_refresh_at: types::Timestamp,
+    /// The UTC timestamp of the broadcaster’s next scheduled ad, in RFC3339 format. Empty if the channel has no ad scheduled or is not live.
+    #[serde(
+        default,
+        deserialize_with = "helix::deserialize_none_from_empty_or_zero_string"
+    )]
+    pub next_ad_at: Option<types::Timestamp>,
+    /// The length in seconds of the scheduled upcoming ad break.
+    pub length_seconds: i32, /* TODO: Is this a types::CommercialLength? is it 0 if no ad is scheduled? */
+    /// The UTC timestamp of the broadcaster’s last ad-break, in RFC3339 format. Empty if the channel has not run an ad or is not live.
+    pub last_ad_at: types::Timestamp,
+    /// The amount of pre-roll free time remaining for the channel in seconds. Returns 0 if they are currently not pre-roll free.
+    #[serde(
+        default,
+        deserialize_with = "helix::deserialize_none_from_empty_or_zero_string"
+    )]
+    pub preroll_free_time_seconds: Option<i32>,
+}
+
+impl Request for GetAdScheduleRequest<'_> {
+    type Response = Option<AdSchedule>;
+
+    const PATH: &'static str = "channels/ads";
+    #[cfg(feature = "twitch_oauth2")]
+    const SCOPE: twitch_oauth2::Validator = twitch_oauth2::validator![any(
+        twitch_oauth2::Scope::ChannelReadAds,
+        twitch_oauth2::Scope::ChannelManageAds
+    )];
+}
+
+impl RequestGet for GetAdScheduleRequest<'_> {
+    fn parse_inner_response(
+        request: Option<Self>,
+        uri: &http::Uri,
+        str_response: &str,
+        status: http::StatusCode,
+    ) -> Result<helix::Response<Self, <Self as Request>::Response>, helix::HelixRequestGetError>
+    where
+        Self: Sized,
+    {
+        let response: Result<helix::InnerResponse<Vec<_>>, _> =
+            crate::parse_json(str_response, true).map_err(|e| {
+                helix::HelixRequestGetError::DeserializeError(
+                    str_response.to_string(),
+                    e,
+                    uri.clone(),
+                    status,
+                )
+            });
+        let response = match response {
+            Ok(resp) => resp,
+            Err(error) => {
+                #[derive(PartialEq, Eq, Deserialize, Default)]
+                struct IsWrong {
+                    snooze_count: i32,
+                    snooze_refresh_at: i32,
+                    next_ad_at: i32,
+                    length_seconds: i32,
+                    last_ad_at: i32,
+                    preroll_free_time_seconds: i32,
+                }
+                let fake: Result<helix::InnerResponse<Vec<IsWrong>>, _> =
+                    crate::parse_json(str_response, true);
+                // if all fields are 0 or we couldn't parse it
+                let empty = match fake {
+                    Ok(fake) if fake.data.first() == Some(&IsWrong::default()) => fake,
```

Comment:
This endpoint is massively wrong right now

https://github.com/twitchdev/issues/issues/857